tests/src/service.spec.ts

```diff
@@ -91,4 +93,58 @@ describe('DebugService', () => {
       expect(modelChangedEvents[0]).to.eq(model);
     });
   });
+
+  describe('protocol', () => {
+    const code = [
+      'i = 0',
+      'i += 1',
+      'i += 1',
+      'j = i**2',
+      'j += 1',
+      'print(i, j)'
+    ].join('\n');
+
+    let breakpoints: Breakpoints.IBreakpoint[];
+    let source_id: string;
+
+    beforeEach(async () => {
+      service.session = session;
+      service.model = model;
+      await service.restoreState(true);
+      const breakpoint_lines: number[] = [3, 5];
+      source_id = service.getCellId(code);
+      breakpoints = breakpoint_lines.map((l: number, index: number) => {
+        return {
+          id: index,
+          line: l,
+          active: true,
+          verified: true,
+          source: {
+            path: source_id
+          }
+        };
+      });
+      await service.updateBreakpoints(code, breakpoints);
+    });
+
+    describe('#updateBreakpoints', () => {
+      it('should update the breakpoints', () => {
+        const bp_list = model.breakpointsModel.getBreakpoints(source_id);
```

Comment:
camelCase for the naming of variables? :)

Reply:
I did too much C++ during the weekend

Reply:
I did too much C++ during the weekend